src/core/Report.cpp

```diff
@@ -23,48 +33,88 @@
 #include <sstream>
 #include <fstream>
 
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <locale>
+#include <iomanip>
+#include <codecvt>
+
+#include "pugixml/pugixml.hpp"
+
+
+template<class Facet>
+struct deletable_facet : Facet
+{
+    template<class ...Args>
+    deletable_facet(Args&& ...args) : Facet(std::forward<Args>(args)...) {}
+    ~deletable_facet() {}
+};
+
 /*
   It writes a header record and a "Start Report" record.
 */
-void CReport::StartReport(LPCTSTR tcAction, const stringT &csDataBase)
+void CReport::StartReport(LPCTSTR tcAction, const stringT &csDataBase, bool writeHeader)
 {
   m_osxs.str(_T(""));
 
   m_tcAction = tcAction;
   m_csDataBase = csDataBase;
 
-  stringT cs_title, sTimeStamp;
-  PWSUtil::GetTimeStamp(sTimeStamp);
-  Format(cs_title, IDSC_REPORT_TITLE1, tcAction, sTimeStamp.c_str());
-  WriteLine();
-  WriteLine(cs_title);
-  Format(cs_title, IDSC_REPORT_TITLE2, csDataBase.c_str());
-  WriteLine(cs_title);
-  WriteLine();
-  LoadAString(cs_title, IDSC_START_REPORT);
-  WriteLine(cs_title);
-  WriteLine();
+  if(writeHeader) {
+    stringT cs_title, sTimeStamp, sAction;
+    PWSUtil::GetTimeStamp(sTimeStamp, true);
+    sAction = _(tcAction);
+    Format(cs_title, IDSC_REPORT_TITLE1, sAction.c_str(), sTimeStamp.c_str());
+    WriteLine();
+    WriteLine(cs_title);
+    Format(cs_title, IDSC_REPORT_TITLE2, csDataBase.c_str());
+    WriteLine(cs_title);
+    WriteLine();
+    LoadAString(cs_title, IDSC_START_REPORT);
+    WriteLine(cs_title);
+    WriteLine();
+  }
+}
+
+static pugi::xml_encoding guessBufferEncoding(uint8_t d0, uint8_t d1, uint8_t d2, uint8_t d3)
```

Comment:
The pugi-namespace is also used in XMPprefs.cpp outside pugixml sub-folder. Pugixml is supporting here. I mislike to encapsulate in a separate class to hide the name space, but if you which it might be possible.

Reply:
pugi:: Namesapce is present also at other places. Encapsulation in a separate class for name space hiding?